src/prognostic_equations/edmfx_closures.jl

```diff
@@ -206,78 +226,193 @@ function mixing_length(
     ᶜstrain_rate_norm,
     ᶜPr,
     ᶜtke_exch,
+    scale_blending_method,
 )
 
     FT = eltype(params)
+    eps_FT = eps(FT)
+
+
     turbconv_params = CAP.turbconv_params(params)
+    sf_params = CAP.surface_fluxes_params(params) # Businger params
+
     c_m = CAP.tke_ed_coeff(turbconv_params)
     c_d = CAP.tke_diss_coeff(turbconv_params)
     smin_ub = CAP.smin_ub(turbconv_params)
     smin_rm = CAP.smin_rm(turbconv_params)
     c_b = CAP.static_stab_coeff(turbconv_params)
     vkc = CAP.von_karman_const(params)
 
-    # compute the maximum mixing length at height z
-    l_z = ᶜz - z_sfc
+    # MOST stability function coefficients
+    most_a_m = sf_params.ufp.a_m # Businger a_m
+    most_b_m = sf_params.ufp.b_m # Businger b_m
+    most_g_m = CAP.coefficient_b_m_gryanik(params)  # Gryanik b_m
 
-    # compute the l_W - the wall constraint mixing length
-    # which imposes an upper limit on the size of eddies near the surface
-    # kz scale (surface layer)
-    if obukhov_length < 0.0 #unstable
-        l_W =
-            vkc * (ᶜz - z_sfc) /
-            max(sqrt(sfc_tke / ustar / ustar) * c_m, eps(FT)) *
-            min((1 - 100 * (ᶜz - z_sfc) / obukhov_length)^FT(0.2), 1 / vkc)
-    else # neutral or stable
-        l_W =
-            vkc * (ᶜz - z_sfc) /
-            max(sqrt(sfc_tke / ustar / ustar) * c_m, eps(FT))
+    # l_z: Geometric distance from the surface
+    l_z = ᶜz - z_sfc
+    # Ensure l_z is non-negative when ᶜz is numerically smaller than z_sfc.
+    l_z = max(l_z, FT(0))
+
+    # l_W: Wall-constrained length scale (near-surface limit, to match 
+    # Monin-Obukhov Similarity Theory in the surface layer, with Businger-Dyer 
+    # type stability functions)
+    tke_sfc_safe = max(sfc_tke, eps_FT)
+    ustar_sq_safe = max(ustar * ustar, eps_FT) # ustar^2 may vanish in certain LES setups
+
+    # Denominator of the base length scale (always positive):
+    #     c_m * √(tke_sfc / u_*²) = c_m * √(e_sfc) / u_*
+    # The value increases when u_* is small and decreases when e_sfc is small.
+    l_W_denom_factor = sqrt(tke_sfc_safe / ustar_sq_safe)
+    l_W_denom = max(c_m * l_W_denom_factor, eps_FT)
+
+    l_W = FT(0)
```

Comment:
I don't think we need this line? We set `l_W` to be something in both `if/else` branches?